manifests/rules/qemu.pp

```diff
@@ -0,0 +1,112 @@
+# @summary Bridged network configuration for qemu/libvirt
+#
+# This class configures the typical firewall setup that libvirt
+# creates. Depending on your requirements you can switch on and off
+# several aspects, for instance if you don't do DHCP to your guests
+# you can disable the rules that accept DHCP traffic on the host or if
+# you don't want your guests to talk to hosts outside you can disable
+# forwarding and/or masquerading for IPv4 traffic.
+#
+# @param interface
+#   Interface name used by the bridge.
+#
+# @param network_v4
+#   The IPv4 network prefix used in the virtual network.
+#
+# @param network_v6
+#   The IPv6 network prefix used in the virtual network.
+#
+# @param dns
+#   Allow DNS traffic from the guests to the host.
+#
+# @param dhcpv4
+#   Allow DHCPv4 traffic from the guests to the host.
+#
+# @param forward_traffic
+#   Allow forwarded traffic (out all, in related/established)
+#   generated by the virtual network.
+#
+# @param internal_traffic
+#   Allow guests in the virtual network to talk to each other.
+#
+# @param masquerade
+#   Do NAT masquerade on all IPv4 traffic generated by guests
+#   to external networks.
+class nftables::rules::qemu (
+  String[1]                               $interface         = 'virbr0',
+  Stdlib::IP::Address::V4::CIDR           $network_v4        = '192.168.122.0/24',
+  Optional[Stdlib::IP::Address::V6::CIDR] $network_v6        = undef,
+  Boolean                                 $dns               = true,
+  Boolean                                 $dhcpv4            = true,
+  Boolean                                 $forward_traffic   = true,
+  Boolean                                 $internal_traffic  = true,
+  Boolean                                 $masquerade        = true,
+) {
+  if $dns {
+    nftables::rule {
+      'default_in-qemu_udp_dns':
+        content => "iifname \"${interface}\" udp dport 53 accept";
```

Comment:
are single quotes allowed as well?
```suggestion
        content => "iifname '${interface}' udp dport 53 accept";
```

I'm not sure if that would make it more readable?

Reply:
mhm, does it need to be quoted at all?

Reply:
The manpage only mentions double quotes:

> Identifiers begin with an alphabetic character (a-z,A-Z), followed zero or more alphanumeric characters (a-z,A-Z,0-9) and the characters slash (/), backslash (\), underscore (_) and dot (.). Identifiers using different characters or clashing with a keyword need to be enclosed in double quotes (").  

```
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft add rule inet filter INPUT iifname iifname accept
Error: syntax error, unexpected iifname
add rule inet filter INPUT iifname iifname accept
                                   ^^^^^^^
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft add rule inet filter INPUT iifname \'iifname\' accept
Error: syntax error, unexpected junk
add rule inet filter INPUT iifname 'iifname' accept
                                   ^
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft add rule inet filter INPUT iifname \"iifname\" accept
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft --version
nftables v0.9.3 (Topsy)
```

Reply:
> mhm, does it need to be quoted at all?

It does to allow interfaces named after reserved words.

Reply:
meh :(